tests/integration/conftest.py

```diff
@@ -214,22 +225,99 @@ async def deploy_model(
 @pytest_asyncio.fixture(scope="module")
 async def kubernetes_cluster(request: pytest.FixtureRequest, ops_test: OpsTest):
     """Deploy local kubernetes charms."""
-    model = "main"
+    cluster_model = "main"
     charm_path = ("worker/k8s", "worker")
     charms = [Charm(ops_test, Path("charms") / p) for p in charm_path]
     charm_files = await asyncio.gather(
         *[charm.resolve(request.config.option.charm_files) for charm in charms]
     )
+
     bundle = Bundle(ops_test, Path(__file__).parent / "test-bundle.yaml")
     for path, charm in zip(charm_files, charms):
         bundle.switch(charm.app_name, path)
-    async with deploy_model(request, ops_test, model, bundle) as the_model:
+
+    async with deploy_model(request, ops_test, cluster_model, bundle) as the_model:
         yield the_model
 
 
 @pytest_asyncio.fixture(scope="module")
+async def cluster_kubeconfig(ops_test: OpsTest, kubernetes_cluster: Model):
+    """Fixture to pull the kubeconfig out of the kubernetes cluster"""
+    k8s = kubernetes_cluster.applications["k8s"].units[0]
+    action = await k8s.run("k8s config")
+    result = await action.wait()
+    assert result.results["return-code"] == 0, "Failed to get kubeconfig with kubectl"
+    kubeconfig_path = ops_test.tmp_path / "kubeconfig"
+    kubeconfig_path.write_text(result.results["stdout"])
```

Comment:
so, i wish this used the juju action instead of running through the snap CLI in order to make use of the public address of the unit

```suggestion
    action = await k8s.run_action("get-kubeconfig")
    result = await action.wait()
    kubeconfig_path = ops_test.tmp_path / "kubeconfig"
    kubeconfig_path.write_text(result.results["kubeconfig"])
```